### PR TITLE
Ha authentication fix

### PR DESCRIPTION
**Pull request by yhwen**
Fixes # .

Fixed the HA abort task issue when SP switching.

### Description

- The authentication check functions are missing during the Cellnet integration migration. This caused the HA SP server switching job run and task abort issue. 
- Added back the authentication functions to the server job process.
- notify the server state change to the running jobs.
- add the GET_CLIENTS_RETRY to 60

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [x] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: nvflare/private/fed/client/client_run_manager.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def __init__(self, job_id):
         self.start_time = None
 
 
-GET_CLIENTS_RETRY = 3
+GET_CLIENTS_RETRY = 60
```
[the remark this answers is not in the document]
Comment (yhwen) [reply nvflare/private/fed/client/client_run_manager.py:54]: under large number of clients, this process could potential take longer time. increase the retry to allow that. The job worker process will wait until this completes then start.

File: nvflare/private/fed/server/fed_server.py [hunk 2]
```diff
@@ -626,6 +626,19 @@ def create_run_manager(self, workspace, job_id):
             handlers=self.runner_config.handlers,
         )
 
+    def authentication_check(self, request: Message, state_check):
+        error = None
+        # server_state = self.engine.server.server_state
+        if state_check.get(ACTION) in [NIS, ABORT_RUN]:
```
[the remark this answers is not in the document]
Comment (yhwen) [reply nvflare/private/fed/server/fed_server.py:632]: Not_In_Service

File: nvflare/private/fed/server/fed_server.py [hunk 3]
```diff
@@ -699,6 +713,27 @@ def _check_server_state(self, overseer_agent):
...
+    def _notify_state_change(self, old_state_name):
+        new_state_name = self.server_state.__class__.__name__
+        if new_state_name != old_state_name:
+            self.logger.info(f"state changed from: {old_state_name} to: {new_state_name}")
+            keys = list(self.engine.run_processes.keys())
+            if keys:
+                target_fqcns = []
+                for job_id in keys:
+                    target_fqcns.append(FQCN.join([FQCN.ROOT_SERVER, job_id]))
+                cell_msg = new_cell_message(headers={}, payload=fobs.dumps(self.server_state))
```
[the remark this answers is not in the document]
Comment (yhwen) [reply nvflare/private/fed/server/fed_server.py:727]: for job_id in keys:
                    target_fqcns.append(FQCN.join([FQCN.ROOT_SERVER, job_id]))

This loop is just create the target_fqcns list. Then it's creating a single cell_msg, and then broadcast_request() to all the targets.

File: nvflare/private/fed/app/simulator/simulator_worker.py [hunk 4]
```diff
@@ -64,6 +64,7 @@ def create_client_runner(self, client):
 
         client_app_runner = SimulatorClientAppRunner()
         client_runner = client_app_runner.create_client_runner(app_client_root, args, args.config_folder, client, False)
+        client_app_runner.sync_up_parents_process(client)
```
Comment (YuanTingHsieh) [inline nvflare/private/fed/app/simulator/simulator_worker.py:67]: Why do we need this line?

File: nvflare/private/fed/server/server_state.py [hunk 5]
```diff
@@ -40,14 +40,14 @@ class ServerState(ABC):
     ABORT_CURRENT_RUN = {ACTION: ABORT_RUN, MESSAGE: "Abort current run"}
     IN_SERVICE = {ACTION: SERVICE, MESSAGE: "Server in service"}
 
+    logger = logging.getLogger("FederatedServer")
```
Comment (YuanTingHsieh) [inline nvflare/private/fed/server/server_state.py:43]: should it be
``logger = logging.getLogger("ServerState")``

File: nvflare/private/fed/app/simulator/simulator_worker.py (hunk 4 above)
Comment (yhwen) [reply nvflare/private/fed/app/simulator/simulator_worker.py:67]: Use the consistent approach as the server to sync up data with the parent process.

File: nvflare/private/fed/server/server_state.py (hunk 5 above)
Comment (yhwen) [reply nvflare/private/fed/server/server_state.py]: fixed.

**Pull request merged**